src/app-constants.js

```diff
@@ -73,4 +79,40 @@ optionalEnvVars.forEach(key => {
   if (value) AppConstants[key] = value
 })
 
+// Create HIBP breach link blocklist
+const linkListData = readFileSync(path.join(
```

Comment:
I’m not sure what we should do for the first iteration before automating things:
1. Add the JSON in this repository as is
2. Add `blurts-https-stats` as a submodule
3. Set `HIBP_BREACH_LINK_BLOCKLIST` directly in `.env`

Currently I’m leaning towards the third options. What do you think @Vinnl, @mansaj and @pdehaan?

Reply:
If 3. is no problem for Brandon, that sounds good to me. Otherwise 1. also sounds absolutely fine to me.

Reply:
@pdehaan It's not super easy by any chance to also output how long ago every breach was whose website did not resolve? That could give us an indication of how soon new breaches are likely to stop resolving.

Reply:
> It's not super easy by any chance to also output how long ago every breach was whose website did not resolve? That could give us an indication of how soon new breaches are likely to stop resolving.

I'd have to think it over. I think my problem is that I'm generally looping over unique `domains`, not breaches. So there isn't always an easy way to find an exact breach by domain (although probably that's just more of an edge case for the [15?] domains that have more than 1 breach).

Reply:
@Vinnl https://github.com/pdehaan/blurts-domains-playwright
Now with tables, so it looks something more like

| STATUS | DOMAIN | BREACH DATE | URL
|:----:|:----|:----:|:----
200/OK | adobe.com | 10/3/2013 | https://www.adobe.com/

Reply:
Awesome! So manually checking the errored ones from 2020 and later, only the shockgore and tamodo domains no longer resolved for me, both from 2020. So I expect we're "safe" for a while if using a "stale" list, and to be extra sure, we could also use a cut-off date of two or one years for when to link to a breached company.

Reply:
But one or two years from _when_? BreachDate? AddedDate? ModifiedDate?
At one point I had a handy calculator that did displayed the number of days between the BreachDate and AddedDate and saw that some breaches weren't added to the HIBP site until like 3-5 years after the BreachDate.

Reply:
https://github.com/pdehaan/blurts-relative-breach-date
TLDR:

- 668 breaches
- AVG DATE DIFF: 472.73 days (1.3 years)
- 174 (26%) have a diff measured in "days" (presumably less than 30 days)
- 243 (36%) have a diff measured in "mos" (greater than 30 days, but less than 12.0 months)
- 248 (37%) have a diff measured in "yrs" (greater than 12 months)

(i'm missing 3 breaches, so there must be a month or year singular that my Ctrl+F isn't picking up -- and I'm too lazy to script/grep it)